src/main/java/org/apache/sling/models/impl/ModelAdapterFactory.java

```diff
@@ -1256,4 +1300,16 @@ private Object handleBoundModelResult(Result<?> result) {
                 scriptEngineFactory, bindingsValuesProvidersByContext).adaptTo(targetClass);
     }
 
+    @Override
+    public void requestDestroyed(ServletRequestEvent sre) {
+        ServletRequest request = unwrapRequest(sre.getServletRequest());
```

Comment:
I also find it weird, that this event does still contained a wrapped servlet request. But unwrapping does not do any harm (in no case).